File: src/WixExtensions/DataSourceExtension/wixext/AccessConnectionInfo.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AppSecInc.Wix.Extensions
{
    /// <summary>
    /// An Access connection info string
    /// </summary>
    public class AccessConnectionInfo
    {
        public string DBQ;
        public string Username;
        public string Password;

        public AccessConnectionInfo()
        {

        }

        /// <summary>
        /// Get the connection string
        /// </summary>
        public string ConnectionString
        {
            get
            {
                StringBuilder sb = new StringBuilder("Driver={Microsoft Access Driver (*.mdb)};");
                sb.AppendFormat("DBQ={0};", DBQ);
                if (! string.IsNullOrEmpty(Username))
                {
                    sb.AppendFormat("Uid={0};Pwd={1};", Username, Password);
                }

                return sb.ToString();
            }
        }
    }
}

File: src/Common/ODBC/AccessConnectionInfo.cpp
#include "StdAfx.h"
#include "AccessConnectionInfo.h"

using namespace AppSecInc::Databases::ODBC;
using namespace AppSecInc::Databases::Access;

AccessConnectionInfo::AccessConnectionInfo(
	const std::wstring& dbq)	
	: _dbq(dbq)
{

}

AccessConnectionInfo::AccessConnectionInfo(const AccessConnectionInfo& info)
{
	operator=(info);
}

void AccessConnectionInfo::operator=(const AccessConnectionInfo& info)
{
	_dbq = info._dbq;
	_username = info._username;
	_password = info._password;
}

std::wstring AccessConnectionInfo::GetConnectionString() const
{
	CHECK_BOOL(! _dbq.empty(), L"DBQ has not been set.");

	std::wstringstream connect_string_s;
	connect_string_s << L"Driver={Microsoft Access Driver (*.mdb)};DBQ=" << _dbq << L";";
	
	if (! _username.empty())
	{
		connect_string_s << L"Uid=" << _username << L";Pwd=" << _password << L";";
	}

	return connect_string_s.str();
}

AccessConnectionInfo::~AccessConnectionInfo()
{

}
